### PR TITLE
feat(image): add image metadata

### PR DESCRIPTION
**Pull request by tonaim**
made below changes.
1. As layer size calculation is not correct. Removed the code related to it.
2. Added changes to get artifact metadata from inspect method.

### CODE REVIEW

File: image/daemon/image.go [hunk 1]
```diff
@@ -92,8 +98,45 @@ func (img *image) ConfigFile() (*v1.ConfigFile, error) {
...
+	loc, err := time.LoadLocation("")
+	if err != nil {
+		return nil, xerrors.Errorf("failed setting location %s: %w", img.inspect.Created, err)
+	}
+	time.Local = loc
+	imgCreatedDate, err := dateparse.ParseLocal(img.inspect.Created)
+	fmt.Println(imgCreatedDate)
+	if err != nil {
+		return nil, xerrors.Errorf("failed parsing created Date %s: %w", img.inspect.Created, err)
+	}
```
Comment (knqyf263) [inline image/daemon/image.go:131]: Could you elaborate on it? Doesn't `time.ParseLocation(img.inspect.Created, time.UTC)` work?

File: types/artifact.go [hunk 2]
```diff
@@ -111,8 +115,15 @@ type ArtifactDetail struct {
...
 	// HistoryPackages are packages extracted from RUN instructions
 	HistoryPackages []Package `json:",omitempty"`
 }
+
+type LayerMetadata struct {
+	Digest           string     `json:",omitempty"`
+	LayerDiffId      string     `json:",omitempty"`
+	UncompressedSize int64      `json:",omitempty"`
+	LayerHistory     v1.History `json:",omitempty"`
+}
```
[the remark this answers is not in the document]
Comment (tonaim) [reply types/artifact.go:129]: I agree history is not needed here. But these are not saved in cache. 

1. From inspect (client) we get v1.config (which will have all metadata like architecture, docker vesion etc). But will not have layer digest
2. So the digest will get it from scan (server) to complete the metadata.  I left the size place holder as it is to use it in future.

File: image/daemon/image.go (hunk 1 above)
Comment (tonaim) [reply image/daemon/image.go:131]: Sure, I assume you are asking about `ParseInLocation`  in https://golang.org/src/time/format.go.

It needs layout as input, that is the problem here. We don't know the date layout of created date.

Comment (knqyf263) [reply image/daemon/image.go]: Why don't we know the date layout?

File: types/artifact.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (knqyf263) [reply types/artifact.go]: `Digest` cannot be obtained when scanning local images even in the server.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**